### DISCUSSION TITLE
Add space or special character in mentions example

### DISCUSSION THREAD

**Discussion by smadapathi** [ianstormtaylor/slate#5098] [Q&A]
Thank you team for this library!
We have requirement like finding user by name(John Doe) or email(john.doe@gmail.com) but in the mentions example, after space or any special character, suggestions popover disappears. Is there any way we can keep suggestions popover even after a `.` or space.

**Comment by dylans** [comment]
The version in Plate (Plugins for Slate) extends the mentions example and has the ability to support special characters. You don't have to use Plate, you could just review the code for its mention component and extend the example as you see fit.

https://github.com/udecode/plate/tree/main/packages/nodes/mention/src

**Reply by smadapathi** [reply]
Thanks @dylans, will take a look. The way we did now is that we are checking if we there are any special character (. or space) in the current word and getting the word before like so:

```
    const [start] = Range.edges(selection);
    const wordBefore = Editor.before(editor, start, { unit: 'word' });
    const before = wordBefore && Editor.before(editor, wordBefore);
    let beforeRange = before && Editor.range(editor, before, start);
    let beforeText = beforeRange && Editor.string(editor, beforeRange);
    if (beforeText?.indexOf('.') !== -1 || beforeText?.indexOf(' ') !== -1) {
      const [startForWordBefore] = Range.edges({
        anchor: {
          offset: before.offset - 1,
          path: before.path,
        },
        focus: {
          offset: before.offset - 1,
          path: before.path,
        },
      });
      const newWordBefore = Editor.before(editor, startForWordBefore, { unit: 'word' });
      const newBefore = newWordBefore && Editor.before(editor, newWordBefore);
      beforeRange = newBefore && Editor.range(editor, newBefore, start);
      beforeText = beforeRange && Editor.string(editor, beforeRange);
    }
```

**Comment by jieming-hu** [comment]
I want to getting the word before like `Alex Macon Michalowicz-Macon`, what should i do to get it, @smadapathi please help me
